contrib/devtools/symbol-check.py

```diff
@@ -176,77 +180,82 @@ def check_version(max_versions, version, arch) -> bool:
 def check_imported_symbols(binary) -> bool:
     ok: bool = True
 
-    for symbol in binary.imported_symbols:
+    for symbol in binary.concrete.imported_symbols:
         if not symbol.imported:
             continue
 
         version = symbol.symbol_version if symbol.has_version else None
 
         if version:
             aux_version = version.symbol_version_auxiliary.name if version.has_auxiliary_version else None
-            if aux_version and not check_version(MAX_VERSIONS, aux_version, binary.header.machine_type):
+            if aux_version and not check_version(MAX_VERSIONS, aux_version, binary.concrete.header.machine_type):
                 print(f'{filename}: symbol {symbol.name} from unsupported version {version}')
                 ok = False
     return ok
 
 def check_exported_symbols(binary) -> bool:
     ok: bool = True
+    if binary.abstract.header.object_type == lief.OBJECT_TYPES.LIBRARY:
+        return ok
```

Comment:
I think one of the most important things to check for shared libraries is that they export the right symbols, and only the right symbols.

Reply:
Yea, that is what I assumed this would be doing when I first looked. Otherwise I think the checks currently being added are ~0.

Reply:
Exactly. It was the initial idea of this PR. When I faced with a crowd of exported symbols, I decided to make this PR focused on introducing a tool, and leave fighting with linker issues for follow ups.